azurerm/resource_arm_app_service_virtual_network_association.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	connectionEnvelope := web.SwiftVirtualNetwork{
+		SwiftVirtualNetworkProperties: &web.SwiftVirtualNetworkProperties{
+			SubnetResourceID: utils.String(d.Get("subnet_id").(string)),
+		},
+	}
+	_, err = client.CreateOrUpdateSwiftVirtualNetworkConnection(ctx, resourceGroup, name, connectionEnvelope)
+	if err != nil {
+		return err
+	}
+	read, err := client.GetSwiftVirtualNetworkConnection(ctx, resourceGroup, name)
+	if err != nil {
+		return err
+	}
+	d.SetId(*read.ID)
+	return resourceArmAppServiceVirtualNetworkAssociationRead(d, meta)
+}
+
+func resourceArmAppServiceVirtualNetworkAssociationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.AppServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["sites"]
+
+	appService, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(appService.Response) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	resp, err := client.GetSwiftVirtualNetworkConnection(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	if resp.SwiftVirtualNetworkProperties == nil {
+		return fmt.Errorf("Error retrieving virtual network properties (App Service %q / Resource Group %q): `properties` was nil", name, resourceGroup)
+	}
+	props := *resp.SwiftVirtualNetworkProperties
+	subnetID := props.SubnetResourceID
+	if subnetID == nil || *subnetID == "" {
+		d.SetId("")
+		return nil
+	}
+	d.Set("subnet_id", subnetID)
+	d.Set("app_service_id", appService.ID)
+	d.Set("resource_group_name", resourceGroup)
+	return nil
+}
+
+func resourceArmAppServiceVirtualNetworkAssociationDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.AppServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Get("app_service_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	subnetID, err := parseAzureResourceID(d.Get("subnet_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["sites"]
+	subnetName := subnetID.Path["subnets"]
+	virtualNetworkName := subnetID.Path["virtualNetworks"]
+
+	locks.ByName(virtualNetworkName, virtualNetworkResourceName)
+	defer locks.UnlockByName(virtualNetworkName, virtualNetworkResourceName)
+
+	locks.ByName(subnetName, subnetResourceName)
+	defer locks.UnlockByName(subnetName, subnetResourceName)
+
+	appService, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(appService.Response) {
+			// assume deleted
+			return nil
```

Comment:
the Read function should have been called prior to the Delete function - meaning this should have been picked up and so this missing would be a bug; as such could we return an error here (e.g. "app service %q not found in resource group %q"?

Reply:
Would read still get called if you ran destroy with `-refresh=false` ?